server/infrastructure/supabase/mappers/AcademicRecordMapper.ts

```
@@ -0,0 +1,129 @@
+    };
+
+    return AcademicRecord.create({
+      studentId,
+      semesters: validatedSemesters,
+      total: validatedSummary,
+      enrollments: [],
+    });
+  }
+
+  static toPersistence(record: AcademicRecord): {
+    semesterRecords: Omit<SemesterRecord, 'id' | 'created_at' | 'updated_at'>[];
```

ESLint

[error] Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.

(@typescript-eslint/array-type)